kubectl-plugin/pkg/cmd/version/version.go

```diff
@@ -25,28 +27,54 @@ func NewVersionOptions(streams genericclioptions.IOStreams) *VersionOptions {
 
 func NewVersionCommand(streams genericclioptions.IOStreams) *cobra.Command {
 	options := NewVersionOptions(streams)
+	// Initialize the factory for later use with the current config flag
 	cmdFactory := cmdutil.NewFactory(options.configFlags)
 
 	cmd := &cobra.Command{
-		Use:   "version",
-		Short: "Output the version of the Ray kubectl plugin and KubeRay operator",
+		Use:               "version",
+		Short:             "Output the version of the Ray kubectl plugin and KubeRay operator",
+		SilenceUsage:      true,
+		ValidArgsFunction: completion.RayClusterCompletionFunc(cmdFactory),
 		RunE: func(cmd *cobra.Command, _ []string) error {
-			fmt.Println("kubectl ray plugin version:", Version)
-
-			kubeClient, err := client.NewClient(cmdFactory)
+			// running cmd.Execute or cmd.ExecuteE sets the context, which will be done by root
+			k8sClient, err := client.NewClient(cmdFactory)
 			if err != nil {
 				return fmt.Errorf("failed to create client: %w", err)
 			}
-
-			operatorVersion, err := kubeClient.GetKubeRayOperatorVersion(cmd.Context())
-			if err != nil {
-				fmt.Println("Warning: KubeRay operator installation cannot be found - did you install it with the name \"kuberay-operator\"?")
-			} else {
-				fmt.Println("KubeRay operator version:", operatorVersion)
-			}
-			return nil
+			return options.Run(cmd.Context(), k8sClient)
```

Comment:
Copied what's in other files like https://github.com/ray-project/kuberay/blob/b2a701d446adbaf0a01d9aa09f8c217e0a839664/kubectl-plugin/pkg/cmd/get/get_cluster.go#L57. Looks like this makes it easier to test.